plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/application/ResourceLabelsPage.java

```diff
@@ -0,0 +1,233 @@
+				.to(BeanProperties.value(IResourceLabelsPageModel.PROPERTY_SELECTED_LABEL).observe(model))
+				.in(dbc);
+		viewer.setContentProvider(new ObservableListContentProvider());
+		viewer.setInput(BeanProperties.list(
+				IResourceLabelsPageModel.PROPERTY_LABELS).observe(model));
+
+		Button addButton = new Button(labelsGroup, SWT.PUSH);
+		GridDataFactory.fillDefaults()
+				.align(SWT.FILL, SWT.FILL).applyTo(addButton);
+		addButton.setText("Add");
+		addButton.addSelectionListener(onAdd());
+		
+		Button editExistingButton = new Button(labelsGroup, SWT.PUSH);
+		GridDataFactory.fillDefaults()
+				.align(SWT.FILL, SWT.FILL).applyTo(editExistingButton);
+		editExistingButton.setText("Edit");
+		editExistingButton.addSelectionListener(onEdit());
+		ValueBindingBuilder
+				.bind(WidgetProperties.enabled().observe(editExistingButton))
+				.notUpdatingParticipant()
+				.to(BeanProperties.value(IResourceLabelsPageModel.PROPERTY_SELECTED_LABEL).observe(model))
+				.converting(new IsNotNullOrReadOnlyBooleanConverter())
+				.in(dbc);
+		
+		Button removeButton = new Button(labelsGroup, SWT.PUSH);
+		GridDataFactory.fillDefaults()
+				.align(SWT.FILL, SWT.FILL).applyTo(removeButton);
+		removeButton.setText("Remove");
+		removeButton.addSelectionListener(onRemove());
+		ValueBindingBuilder
+				.bind(WidgetProperties.enabled().observe(removeButton))
+				.notUpdatingParticipant()
+				.to(BeanProperties.value(IResourceLabelsPageModel.PROPERTY_SELECTED_LABEL).observe(model))
+				.converting(new IsNotNullOrReadOnlyBooleanConverter())
+				.in(dbc);
+	}
+
+	private Label getSelectedEntry() {
+		return (Label) viewer.getStructuredSelection().getFirstElement();
+	}
+	
+	private SelectionListener onRemove() {
+		return new SelectionAdapter() {
+
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				Label label = getSelectedEntry();
+				if(MessageDialog.openQuestion(getShell(), "Remove Label", NLS.bind("Are you sure you want to delete the label {0} ", label.getName()))) {
+					model.removeLabel(label);
+					viewer.refresh();
+				}
+			}
+			
+		};
+	}
+
+	private SelectionListener onEdit() {
+		return new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				final Label label = getSelectedEntry();
+				IKeyValueWizardModel<Label> dialogModel = new KeyValueWizardModelBuilder<Label>(label)
+						.title("Edit Label")
+						.description("Edit the resource label.")
+						.keyLabel("Label")
+						.groupLabel("Label")
+						.keyAfterConvertValidator(null)
+						.valueAfterConvertValidator(null)
+						.build();
+				OkCancelButtonWizardDialog dialog =
+						new OkCancelButtonWizardDialog(getShell(),
+								new KeyValueWizard<Label>(getSelectedEntry(), dialogModel));
+				if(OkCancelButtonWizardDialog.OK == dialog.open()) {
+					model.updateLabel(label, dialogModel.getKey(), dialogModel.getValue());
+					viewer.refresh();
```

Comment:
This refresh wouldnt be required if your label class would fire change events, the binding would refresh the viewer.